### DISCUSSION TITLE
Inquiry about the flux in x direction

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#22542] [Q&A Modules: Phase field] [answered]
Hi everyone,

I was wondering if there is flux in x direction in 2D model. Since the model is 2D, the flux in x direction means the flux that passes through the y-z plane, however, there is no z axis in 2D model, Why is there still flux in x direction? Is my way of calculating the jx not right?
This is how I calculate flux in x direction:
![image](https://user-images.githubusercontent.com/63679626/198906064-c4d9ee4c-591f-4907-b531-72258da03629.png)
(Mobility=1 here)
The following is jx:
![image](https://user-images.githubusercontent.com/63679626/198906243-e03de4f7-03f4-4c4f-b86b-24691b179f91.png)

Thank you,
Best,

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you please paste your mesh block here?
I suspect your mesh is actually in the x,y plane?

Guillaume

**Reply by abc-hy** [reply]
![image](https://user-images.githubusercontent.com/63679626/199023212-fc86aa3f-bd9f-408e-a2b4-dc581d93a6e6.png)
Yes. only in x,y plane. no z axis.
Why do I get flux in x direction?

**Reply by GiudGiud** [reply]
because the simulation is in the xy plane?
in the Z direction (out of plane) I would expect 0 flux

**Reply by abc-hy** [reply]
But jx must be normal to y-z plane, if there is no z axis, there is no that plane. 
Yes, there is no z axis, so, there is 0 flux in z direction.

**Reply by abc-hy** [reply]
Where is wrong?

**Reply by GiudGiud** [reply]
your mesh is in the XY plane not the XZ

**Reply by abc-hy** [reply]
????what? can you explain it clearly?

**Reply by GiudGiud** [reply]
nz = 0 there are no elements in the Z direction

**Reply by abc-hy** [reply]
yes. So, there is no z axis, then, why do I get flux in x direction?

**Reply by GiudGiud** [reply]
I think you should explain why there should not be flux in the X direction

**Reply by abc-hy** [reply]
because based on the the definition of flux, flux unit, which is #/(m^2*s), it means the amount of solute that pass through per unit area unit time, since , there is no z axis, so there is no y-z plane, ( x direction is normal to y-z plane), if there is no y-z plane, then where does the flux come from?

**Reply by GiudGiud** [reply]
no a 2D plane is not an infinitely thin simulation here, it's the contrary, it's an infinitely thick simulation.
It's as if it had a length of 1 in the Z direction, with reflective (0 flux) boundary conditions on the top and bottom Z planes

**Reply by abc-hy** [reply]
Is the infinitely thick simulation a default setting？
Could you explain "with reflective (0 flux) boundary conditions on the top and bottom Z planes"?
What is top and bottom Z planes? Z is an axis, right?

**Reply by GiudGiud** [reply]
By Z plane I meant planes with a Z-axis normal
How would you do infinitely thin?

**Reply by abc-hy** [reply]
thickness equals to 0, meaning that there is no z axis.

**Reply by GiudGiud** [reply]
you can try a 3D simulation with a non zero width then progressively reduce it to 0

**Reply by abc-hy** [reply]
Do you mean the 2D model in MOOSE are all actually 3D model???

**Reply by GiudGiud** [reply]
no

**Reply by abc-hy** [reply]
Although I did not set length on z axis, the MOOSE itself consider it a 3D model?
 You said: a 2D plane is not an infinitely thin simulation here, it's the contrary, it's an infinitely thick simulation.
COuld you please explain this?

**Reply by abc-hy** [reply]
Then could you tell me, what is 2D model in MOOSE? Is there thickness on the z axis?

**Reply by GiudGiud** [reply]
No it's a 2D model.
You should think about what it means to do a 2D simulation. It usually assumes infinite periodicity in the 3rd dimension

**Comment by abc-hy** [comment]
infinite periodicity? What periodicity? I did not set periodic boundary
condition, I used zero flux BC

**Reply by GiudGiud** [reply]
for the 3rd dimension, you likely effectively have 0 flux which is equivalent to periodic.

This is dependent on what was implemented in the kernels. For reactor physics, you can implement 2D systems with an axial model.

**Reply by abc-hy** [reply]
It seems that 2D model and 3D model does not have difference. Could you please clarify the difference of 2D model and 3D model?

**Reply by GiudGiud** [reply]
In a 3D model you may have heterogeneity in the 3rd dimension, in 2D you cant
